background/processor.js

```diff
@@ -55,6 +66,30 @@ export default class Processor {
     return lookup.address(address);
   }
 
+  cleanImages (data) {
+    const omit = (data) => {
+      return data instanceof Promise || (Date.now() - data.date) > IMAGES_TTL;
+    };
+
+    return omitBy(data, omit);
+  }
+
+  loadImages () {
+    chrome.storage.local.get(IMAGES_STORAGE_KEY, (storage = {}) => {
+      const images = storage[IMAGES_STORAGE_KEY] || {};
+
+      // Load the saved images, omitting the old data
+      this._images = this.cleanImages(images);
+      this.saveImages();
+    });
+  }
+
+  saveImages () {
+    const data = this.cleanImages(this._images);
```

Comment:
I think it might be good to do saving asynchronously (wrap into `setImmediate/setTimeout(0)`), but it will most probably be noticable only for a super large collections (i.e. long running background script)